### PR TITLE
Implement basket api

### PR DESCRIPTION
**Pull request by HaidukEvgen**

### CODE REVIEW

File: OnlineShop.Services.Basket.BusinessLayer/Services/Implementations/BasketService.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            {
+                throw new BasketNotFoundException(userId);
+            }
+
+            var orderCreateDto = new OrderCreateDto
+            {
+                UserId = userId,
+                ProductIds = basket.Items.Select(basketItem => basketItem.ProductId).ToArray(),
+                Total = basket.TotalPrice,
+                Comment = orderDetailsDto.Comment,
```
Comment (katyalys) [inline OnlineShop.Services.Basket.BusinessLayer/Services/Implementations/BasketService.cs]: try to use conditional mapping here

File: OnlineShop.Services.Basket.BusinessLayer/Services/Implementations/BasketService.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                DeliveryAddress = orderDetailsDto.DeliveryAddress,
+                PhoneNumber = orderDetailsDto.PhoneNumber
+            };
+
+
+            //TODO: Replace this temporary synchronous communication with RabbitMQ message bus
+            var apiUrl = "https://localhost:7004/api/orders";
+
```
Comment (katyalys) [inline OnlineShop.Services.Basket.BusinessLayer/Services/Implementations/BasketService.cs:71]: create separate method

File: OnlineShop.Services.Basket.DataLayer/Models/Data/ShoppingCart.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    public class ShoppingCart
+    {
+        public List<ShoppingCartItem> Items { get; set; } = new List<ShoppingCartItem>();
+
+        public decimal TotalPrice
+        {
+            get
+            {
+                decimal totalprice = 0;
+                foreach (var item in Items)
```
Comment (katyalys) [inline OnlineShop.Services.Basket.DataLayer/Models/Data/ShoppingCart.cs:13]: empty line before loop

File: OnlineShop.Services.Basket.DataLayer/Models/Data/ShoppingCart.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+                {
+                    totalprice += item.Price * item.Quantity;
+                }
+                return totalprice;
+            }
```
Comment (katyalys) [inline OnlineShop.Services.Basket.DataLayer/Models/Data/ShoppingCart.cs:19]: add empty line before return

File: OnlineShop.Services.Catalog.Infrastructure/Repositories/Implementations/ProductRepository.cs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        private readonly ICatalogContext _catalogContext;
+
+        public ProductRepository(ICatalogContext catalogContext)
+        {
+            _catalogContext = catalogContext;
+        }
+
+        public async Task<IEnumerable<Product>> GetAllAsync(CancellationToken cancellationToken)
+        {
```
Comment (katyalys) [inline OnlineShop.Services.Catalog.Infrastructure/Repositories/Implementations/ProductRepository.cs:19]: implement pagination

**Review by katyalys — CHANGES_REQUESTED** [review]

File: OnlineShop.Services.Basket.BusinessLayer/Services/Implementations/BasketService.cs (hunk 2 above)
Comment (HaidukEvgen) [reply OnlineShop.Services.Basket.BusinessLayer/Services/Implementations/BasketService.cs:71]: later on i will get rid of this code and publish message to rabbitmq instead

**Review by katyalys — APPROVED** [review]

**Pull request merged**